libvast/builtins/operators/from_read.cpp

```diff
@@ -0,0 +1,185 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2023 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <vast/concept/parseable/string.hpp>
+#include <vast/concept/parseable/vast/identifier.hpp>
+#include <vast/concept/parseable/vast/pipeline.hpp>
+#include <vast/error.hpp>
+#include <vast/legacy_pipeline.hpp>
+#include <vast/logger.hpp>
+#include <vast/plugin.hpp>
+
+#include <arrow/type.h>
+#include <caf/error.hpp>
+
+namespace vast::plugins::from_read {
+
+namespace {
+
+class from_operator final : public crtp_operator<from_operator> {
+public:
+  explicit from_operator(const loader_plugin& loader, record config)
+    : loader_plugin_{loader}, config_{std::move(config)} {
+  }
+
+  auto operator()(operator_control_plane& ctrl) const
+    -> caf::expected<generator<chunk_ptr>> {
+    return loader_plugin_.make_loader(config_, ctrl);
+  }
+
+  auto to_string() const -> std::string override {
+    return fmt::format("from {} <{}>", loader_plugin_.name(), config_);
+  }
+
+private:
+  const loader_plugin& loader_plugin_;
+  record config_;
+};
+
+class read_operator final : public crtp_operator<read_operator> {
+public:
+  // TODO: Replace `plugin` with `parser_plugin` (once we have it).
+  explicit read_operator(const plugin& parser, record config)
+    : parser_plugin_{parser}, config_{std::move(config)} {
+  }
+
+  auto operator()(generator<chunk_ptr>, operator_control_plane&) const
+    -> caf::expected<generator<table_slice>> {
+    // TODO: Implement this.
+    return caf::make_error(ec::unimplemented,
+                           "parser_plugin does not exist yet");
+  }
```

Comment:
Please write a follow-up issue for doing this once we have the JSON parser so we can track this somewhere.